crates/cubecl-core/src/ir/processing.rs

```diff
@@ -184,6 +184,37 @@ impl ScopeProcessing {
                     sanitize_constant_scalar_ref_var(&mut op.lhs, &op.out);
                     sanitize_constant_scalar_ref_var(&mut op.rhs, &op.out);
                 }
+                Operator::Bitcast(_) => {}
+                Operator::AtomicLoad(_) => {}
+                Operator::AtomicStore(_) => {}
+                Operator::AtomicSwap(op) => {
+                    sanitize_constant_scalar_ref_var(&mut op.rhs, &op.out);
+                }
+                Operator::AtomicCompareAndSwap(op) => {
+                    sanitize_constant_scalar_ref_var(&mut op.cmp, &op.out);
+                    sanitize_constant_scalar_ref_var(&mut op.val, &op.out);
+                }
+                Operator::AtomicAdd(op) => {
+                    sanitize_constant_scalar_ref_var(&mut op.rhs, &op.out);
+                }
+                Operator::AtomicSub(op) => {
+                    sanitize_constant_scalar_ref_var(&mut op.rhs, &op.out);
+                }
+                Operator::AtomicMax(op) => {
+                    sanitize_constant_scalar_ref_var(&mut op.rhs, &op.out);
+                }
+                Operator::AtomicMin(op) => {
+                    sanitize_constant_scalar_ref_var(&mut op.rhs, &op.out);
+                }
+                Operator::AtomicAnd(op) => {
+                    sanitize_constant_scalar_ref_var(&mut op.rhs, &op.out);
+                }
+                Operator::AtomicOr(op) => {
+                    sanitize_constant_scalar_ref_var(&mut op.rhs, &op.out);
+                }
+                Operator::AtomicXor(op) => {
+                    sanitize_constant_scalar_ref_var(&mut op.rhs, &op.out);
```

Comment:
Is there a reason not to sanitize lhs as well?

Reply:
As I understood it the santization is to ensure matching types, but `lhs` will explicitly not match (`AtomicUInt` vs `AtomicI32`). It can also be either of those. That being said, I can probably use the `Primitive` associated type to do the santization. I'll give it a try.

Reply:
Nevermind, I no longer have the type information at that point. So unless I match on every possible atomic variant I can't really sanitize that part.